ERCS/erc-7579.md

```diff
@@ -215,6 +215,16 @@ This standard separates modules into the following different types that each has
 - Fallback (type id: 3)
 - Hooks (type id: 4)
 
+**Dynamic Module Type Identification and Interaction**
+
+- Smart accounts SHALL employ a dynamic system to identify module types, allowing for the addition of new module types without altering the core standard.
+- Module interactions SHALL be defined to allow behavior modification based on module type, ensuring adaptability to evolving needs.
```

Comment:
What exactly do you mean here?

Reply:
I appreciate the current structure of ERC-7579, but it could benefit from a more dynamic approach to accommodate new module types. 
Currently, the standard is somewhat opinionated, focusing on specific module types like Validators and Executors. To enhance flexibility, we could consider implementing more generic functions for module management. 

This approach would simplify the integration of unforeseen module types in the future. Here's a concept:

```solidity
function installModule(address module, uint256 moduleType, bytes calldata data) external;
function uninstallModule(address module, uint256 moduleType, bytes calldata data) external;
```

Is it practical and helpful for the standard adaptability?

Reply:
Yes, this was something that we considered early on but ultimately decided against for several reasons:

- the feedback we got was that even though this solution looks cleaner, it is less verbose which makes it harder to reason about it
- because new module types cannot just be added without changes to existing accounts, there is not much upside in making the install/uninstall function this flexible apart from 2 fewer functions

@zeroknots anything I missed?

Reply:
You could also easily scope the different function sig in a validator that allows `enableValidator`. Which could be very useful for recovery mechanisms.

furthermore, not having a condition in the function is better on gas